cpp/src/arrow/sparse_tensor.h

```diff
@@ -292,35 +339,81 @@ class SparseTensorImpl : public SparseTensor {
                    const std::vector<std::string>& dim_names = {})
       : SparseTensorImpl(NULLPTR, type, NULLPTR, shape, dim_names) {}
 
-  /// \brief Construct a sparse tensor from a dense tensor
+  /// \brief Create a SparseTensor with full parameters
+  static Result<std::shared_ptr<SparseTensorImpl<SparseIndexType>>> Make(
+      const std::shared_ptr<SparseIndexType>& sparse_index,
+      const std::shared_ptr<DataType>& type, const std::shared_ptr<Buffer>& data,
+      const std::vector<int64_t>& shape, const std::vector<std::string>& dim_names) {
+    if (!is_tensor_supported(type->id())) {
```

Comment:
Is it useful to inline the implementation here? Otherwise, it's clearer if the header file only contains declarations.

Reply:
I'll make it inline.

I guess we will get trouble on mingw32 platform if we leave the implementation of this function out from the header file because the class is template.

Reply:
Ah, I had forgotten about the mingw32 issues. Thanks :-)